Updates Astro Framework Guide for deploying on Workers (#21114)
cloudflare/cloudflare-docs: src/content/docs/workers/frameworks/framework-guides/astro.mdx

</TabItem>

<TabItem label="CLI">

Scaffold an Astro project on Workers, and pick your template.

<PackageManagers
	type="create"
	pkg="cloudflare@latest"
	args="my-astro-app -- --framework=astro --platform=workers"
/>

</TabItem>
</Tabs>

</Aside>

## What is Astro?

[Astro](https://astro.build/) is a web framework built with JavaScript and designed for creating websites that display large amounts of content (such as blogs, documentation sites, or online stores). Astro emphasizes performance through minimal client-side JavaScript, as well as other [design principles](https://docs.astro.build/en/concepts/why-astro/#design-principles). By default, it renders most content on the server, and only the parts of a page requiring interactivity are shipped to the browser. See [Key features](/workers/frameworks/framework-guides/astro/#astros-key-features) for more information about Astro, and check out their [integrations](https://astro.build/integrations/) to see how to add functionality from community integrations.

> ## Astro's Key features

Reviewer: I wonder if we could condense down the `What is Astro` and `Astro's Key features` into one paragraph? I think most people would have chosen a framework and _then_ come to this page, and so probably don't need to be told about Astro. And if someone wanted a lot of detail about Astro, they'd probably go to the Astro docs instead.

I definitely have a personal preference for concise docs though, so feel free to ignore this! 🧡 

Something like:

[Astro](https://astro.build/) is a JavaScript web framework designed for creating websites that display large amounts of content (such as blogs, documentation sites, or online stores). Astro emphasizes performance through minimal client-side JavaScript - by default, it renders as much content as possible at build time, or on demand on the server, which can be a Cloudflare Worker.  [“Islands”](https://docs.astro.build/en/concepts/islands/) of JavaScript are added only where interactivity or personalization is needed.

Astro is also **framework-agnostic**, and supports every major UI framework, including React, Preact, Svelte, Vue, SolidJS , via its official [integrations](https://astro.build/integrations/).

#### Islands

Most of an Astro page is delivered as static HTML, with small [“islands”](https://docs.astro.build/en/concepts/islands/) of JavaScript added only where interactivity or personalization is needed. These can be client or server islands:

- [**Client islands**](https://docs.astro.build/en/concepts/islands/#client-islands) run in the browser and handle interactive features.
- [**Server islands**](https://docs.astro.build/en/concepts/islands/#server-islands) can offload certain computations or data fetching to the server, minimizing client work.

#### UI-agnostic

Astro is **framework-agnostic**, and supports every major UI framework, including React, Preact, Svelte, Vue, SolidJS , via its official [integrations](https://astro.build/integrations/).

## Deploy a new Astro project on Workers

<Steps>
1. **Create a new project with the create-cloudflare CLI (C3).**
    <PackageManagers
    	type="create"
    	pkg="cloudflare@latest"
    	args="my-astro-app -- --framework=astro --platform=workers"
    />

     <Details header="What's happening behind the scenes?">

>      When you run this command, C3 creates a new project directory, initiates [Astro's official setup tool](https://docs.astro.build/en/tutorial/1-setup/2/), and provides the option to deploy instantly.

Reviewer 2: The phrase "provides the option" confused me at first.

I realise after looking at it a few times you are saying that the final step of running this command is that the user is presented with the option to deploy the Worker. Right?

For me it would be a bit clearer if it was something like:

```suggestion
     When you run this command, C3 creates a new project directory, initiates [Astro's official setup tool](https://docs.astro.build/en/tutorial/1-setup/2/), and configures the project for Cloudflare. It then offers the option to deploy the Worker instantly to Cloudflare.
```

    </Details>

2.  **Develop locally.**

        After creating your project, run the following command in your project directory to start a local development server.

    <PackageManagers type="run" args="dev" />

3.  **Deploy your project.**

        You can deploy your project to a [`*.workers.dev` subdomain](/workers/configuration/routing/workers-dev/) or a [custom domain](/workers/configuration/routing/custom-domains/) from your local machine or any CI/CD system (including [Workers Builds](/workers/ci-cd/#workers-builds)). Use the following command to build and deploy. If you're using a CI service, be sure to update your "deploy command" accordingly.

        <PackageManagers type="run" args="deploy" />

  </Steps>

## Deploy an existing Astro project on Workers

<Steps>
1. **Install the Astro adapter.**
    <PackageManagers
    	type="exec"
    	pkg="astro"
    	args="add cloudflare"
    />

     <Details header="What's happening behind the scenes?">

>      This command installs the adapter and makes the appropriate changes to your `astro.config.mjs` file in one step. You can read more about the adapter configuration options [here](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#options).

Reviewer 2: ```suggestion
     This command installs the Cloudflare adapter and makes the appropriate changes to your `astro.config.mjs` file in one step. You can read more about the adapter configuration options [here](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#options).
```

    </Details>

2.  **Build your project.**

    <PackageManagers type="run" args="build" />

3.  **Deploy your project.**

>         <PackageManagers type="run" args="deploy --assets=dist" />

Reviewer: i think we should recommend people create a wrangler config file, instead of using the cli arg? 

and I think you'll have to build before deploying, probably worth advising people to add something to their package json so they can just run `npm run deploy`

for reference these are the scripts from the package.json that c3 generates for astro

	"scripts": {
		"dev": "astro dev",
		"build": "astro build",
		"preview": "astro build && wrangler dev",
		"deploy": "astro build && wrangler deploy",
		"cf-typegen": "wrangler types"
	}

---

import {
	Aside,
	Badge,
	Details,
	Steps,
	WranglerConfig,
	Description,
	InlineBadge,
	Render,
	TabItem,
	Tabs,
	PackageManagers,
} from "~/components";

<Aside type="tip" title="Get started">

<Tabs syncKey="DeployButtonSnippet"> <TabItem label="Dashboard">

Create a blog with Astro and deploy it on Cloudflare Workers as a static website.

> [![Deploy to Workers](https://deploy.workers.cloudflare.com/button)](https://dash.cloudflare.com/?to=/:account/workers-and-pages/create/deploy-to-workers&repository=https://github.com/cloudflare/templates/tree/staging/astro-blog-starter-template)

Author: Quick edit: Instead of using tabs, got feedback that we should just list out the options -- in this case, it would make sense to first present the CLI option (since you can choose your template), and then off a "Deploy to Workers" button to deploy a blog starter template on the user's account.

Reviewer: This okay?

<img width="739" alt="Screenshot 2025-03-27 at 14 17 06" src="https://github.com/user-attachments/assets/0d60167d-4eda-4f32-ada4-54911597b3ee" />

import {
	Badge,
	Details,
	Steps,
	WranglerConfig,
	Description,
	InlineBadge,
	Render,
	TabItem,
	Tabs,
	PackageManagers,
} from "~/components";

**Start from CLI** - scaffold an Astro project on Workers, and pick your template.

<PackageManagers
	type="create"
	pkg="cloudflare@latest"

> 	args="my-astro-app -- --framework=astro --platform=workers"

Reviewer 3: It looks like this change generate an invalid command:

```plain
npm create cloudflare@latest -- my-astro-app -- --framework=astro --platform=workers
```

The first `--` should not be there

Reviewer 2: That sounds right. I wonder if this helper is clever enough to add the `--` when needed?

/>

---

> **Or just deploy** - create a static blog with Astro and deploy it on Cloudflare Workers.

Reviewer 4: ```suggestion
**Or just deploy** - create a static blog with Astro and deploy it on Cloudflare Workers. We'll create a repository, and set up CI/CD and and preview deployments for you, via [Workers Builds](/workers/ci-cd/builds/).
```

@korinne @nevikashah @kale-stew - I'm not sure what I'm suggesting is quite right, but directionally — it seems like we have an opportunity to tell people that if they choose this path (here or anywhere else) that they'll get this all setup for them?

Just wondering if people already intuit this or not. Maybe overkill? Your call but just think there is so much you get with going down this path that I'm not _quite_ sure people know is possible otherwise?
